### ISSUE TITLE
bounty for removing/improving "team requirement"

### ISSUE THREAD

**Issue by Erkan-Yilmaz** [Erkan-Yilmaz/Gridcoin-tasks#94]
bounty is not certain yet, only discussed

further info on topic: https://github.com/gridcoin/Gridcoin-Research/issues/212

mentioned in [hangout 26](https://github.com/Erkan-Yilmaz/Gridcoin-hangout-minutes/blob/master/hangout_2017_03_18.MD): making a bounty for removing/improving "team requirement"

**Comment by grctest** [comment]
Related: https://github.com/gridcoin/Gridcoin-Research/issues/212

I pledge 50k GRC to anyone that can implement a DPOR/NN solution which is capable of supporting the entire BOINC network (250k to 4 million users) in a decentralized/distributed manner.

We could potentially bring foundation funds into the mix if this proves to be a significant amount of work?

**Comment by gustav1976** [comment]
yes, this should be in the interest of us all, so let's incorporate foundation funds. I'd vote in favour of that.

**Comment by Erkan-Yilmaz** [comment]
[lennstar](https://steemit.com/@lennstar) wrote [here](https://steemit.com/gridcoin/@erkan/get-bounties-for-helping-team-gridcoin-v2#@lennstar/re-erkan-get-bounties-for-helping-team-gridcoin-v2-20170320t170602794z):
"If supporting means in the normal client and without overloading a normal user or project, then I put in 20K too."

**Comment by Erkan-Yilmaz** [comment]
Vortac adds another 30k GRC [here](https://cryptocointalk.com/topic/1331-new-coin-launch-announcement-grc-gridcoin/page-1231#entry235612):
* "However, if we go the bounty way, then I pledge another 30k GRC."

and also says:
* "I think we should have a Foundation expense (and vote) for this. Would give more weight to it and become a final stance on the matter. After all, if this is implemented, it will affect the whole network (ultimately positively, I hope)."

**Comment by Erkan-Yilmaz** [comment]
from [ALIEN](https://cryptocointalk.com/user/161-a-l-i-e-n/), [here](https://cryptocointalk.com/topic/1331-new-coin-launch-announcement-grc-gridcoin/page-1231#entry235640):
"I pledge **100K** if you guys think the bounty approach will be fruitful.  I don't think it would be an issue for the foundation to cover any remaining amount after community donations for a total 1 mil GRC bounty.  It would be a large project and the reward should reflect the caliber of work we would be expecting.  There could maybe be an additional reward for providing 6 months of support or something to ensure it transitions smoothly."

**Comment by grctest** [comment]
If we can harden proof of CPID ownership, and reduce the amount of data that we would be downloading from project web servers (less than the full user.xml.gz file which has irrelevant data) we could potentially remove the mandatory team requirement with minimal changes to the neural network : https://github.com/Erkan-Yilmaz/Gridcoin-tasks/issues/101

**Comment by grctest** [comment]
https://github.com/gridcoin/Gridcoin-Research/issues/289

**Comment by grctest** [comment]
Just to confirm, regardless of the trading rate of GRC my 50k GRC commitment to this issue stands, it's one of the highest priority issues IMO.

Like/Retweet: https://twitter.com/customminer/status/862633377689726976

**Comment by Vortac2** [comment]
Absolutely, my commitment is firm as well. I would support a new Foundation expense too.

**Comment by Erkan-Yilmaz** [comment]
from chat:

stevils:  I was thinking about the expansion to the rest of the BOINC community.  Could adding non-Gridcoin teams as a new pool be implemented more easily than adding individuals?
Erkan:  would they then really come ?
Erkan:  a pool means losing attribution of name to the work done
stevils:  If they want full benefits to stake alone then they join the team but we can open up teams as pools
stevils:  then they join our team

**Comment by mistermarmot** [comment]
Team loyalty (tribalism oxytocin function) is a major impediment to getting people to leave the team they were with for years and join us even with profit motive.

As a first step of implementation, I was suggesting to Erkan in IRC, that we allow teams to join as pools.  
It maybe that no team will join, but of all the teams, it seems likely that many might want to earn some coin even if it means giving up individual contribution RAC as long as they can see their team possibly beat our pools or high ranking individuals.   Many teams would love to get the epeen of stomping us on our 'own field', but yet GRC still gets loyal adopters and spreads it's circulation to more people.

We will likely gain new Gridcoin users from their team mates that want their individual RAC back.

My coding abilities are 30 years out of date so, is this more easily accomplished than the individual, teamless, implementation?

**Comment by Foggyx420** [comment]
Iam a little confused as to the direction you are speaking of @mistermarmot . Please expand on your comment.

My thoughts:
if we allow teams as pools it still would need the team requirement part removed.
also a Pool would still require a single CPID for a wallet and that wallet paying out everyone and that would be the team trusting that one person to do right. A pool is a group of users working for a single wallet.

Regardless team requirement removal would still allow all users going for gridcoin to remain in there own choice of teams or no team at all. I think this is the better option.

**Comment by RobertoGaudio** [comment]
I've thought about it for a while and came up with this possible solution: how about not changing Gridcoin but BOINC? By letting the BOINC user join two teams instead of one people could easily stay in their beloved team and join Team Gridcoin beside it. This way they can continue to crunch for their teams and earn some GRC beside it. 
I think that this would lift up the team requirement barrier for a hell of a lot dedicated crunchers. What do you all think about it?

Regards
RobertoGaudio aka. Nuad

**Comment by gregagnew** [comment]
What if we have each user be responsible for looking up, and pushing their own accounts into the block chain as a transaction to be verified by others using -> http://boinc.berkeley.edu/trac/wiki/WebRpc

So it would work like this:

User downloads boinc & signs up for projects with e-mail.
User downloads gridcoin wallet & attaches same e-mail
User selects projects in gridcoin wallet that they are attached to, and the md5d passwords associated
Users wallet queries the urls with WebRPC, and their e-mail + m5'd password
User adds POR transaction to block chain with (receiving address, email, md5, project, total credits done), to be confirmed by other wallets.

There would need to be a central node still for e-mail -> receiving GRC address confirmation

So:

- User receives POR block
- User verifies e-mail -> recieving address from gridcoin.us
- User gets XML from project using email/md5 and verifies work done
- User confirms block

**Comment by Slapbox** [comment]
@RobertoGaudio We have to focus on what's within our control. I don't think BOINC is going to change this. It's been this way for 20 years and working well for every purpose but mining GRC.

**Comment by gregagnew** [comment]
https://github.com/gridcoin/Gridcoin-Research/issues/341

If you look at my comment here, I've updated the DPoR mechanism proposed.

**Comment by iamtheghostlove** [comment]
I've made a commit to the BOINC github to resolve this issue. I hope the bounty is still open.

**Comment by gregagnew** [comment]
Oops that was my other account. I've made a commit the BOINC github to resolve this issue. After it is accepted I'll make the necessary changes to this repository.

**Comment by grctest** [comment]
@RobertoGaudio I think that adding a second team to a BOINC account would break a lot of existing 3rd party BOINC statistics websites, and would require all BOINC projects to inherit said change, as opposed to the Gridcoin network changing.

@gregagnew 
> (receiving address, email, md5, project, total credits done)

The email and md5 of the password are confidential and shouldn't be shared with the network. If I have your password's md5, I can login to grab your account key and have established a permanent BOINC account compromise.

**Comment by gregagnew** [comment]
I completely agree, that is why I have modified the BOINC API to accept userid's and establish credit. Those commits are not yet pulled, but once they are, we should be able to get users credit with only their e-mail!

The team requirement would no longer be necessary.

The new API feature will allow E-mail -> UserID and then UserID->Credits. Which by-passes the 'get_team_list' call.

A central service is still required (as it is now) to associate e-mails with GRC addresses.

**Comment by RobertoGaudio** [comment]
@gregagnew : i'm happy that my idea seem to have deduced to a possible solution. :)

**Comment by grctest** [comment]
@gregagnew So you're proposing to make both the user's email address and password publicly available on the blockchain? This isn't a secure proposal. Email address use within the Gridcoin client is currently entirely private, email addresses are not publicly published. Making your email address publicly known [is massively insecure](https://github.com/Erkan-Yilmaz/Gridcoin-tasks/issues/106).

We don't need to change the reward mechanism like you're proposing to remove the team requirement. Upgrades to the neural network mechanism, removing the team check and hardening the proof of CPID ownership are in scope of this bounty. (At least for my portion).

**Comment by gregagnew** [comment]
Okay, I'm not proposing any of those things. I did propose those things as a potential work around to the lack of BOINC development. I understand they are insecure. I have since taken the BOINC development upon myself.

Forget about anything I said before today. 

Nothing is required to be on the blockchain, not email or password.

This is all I am proposing:
The new API feature will allow E-mail -> UserID and then UserID->Credits. Which by-passes the 'get_team_list' call.

**Comment by gregagnew** [comment]
Wait, this entire thread is about removing the team requirement.

"200k GRC bounty -> for removing/improving "team requirement" "

Its literally the name of the thread.

**Comment by grctest** [comment]
> Wait, this entire thread is about removing the team requirement.
>
> "200k GRC bounty -> for removing/improving "team requirement" "
>
> Its literally the name of the thread.

That may be the case, but you must take into account the contents of the thread which lay out the requirements for the bounty.

----

>Nothing is required to be on the blockchain, not email or password.
>> * UserA (Minter) adds POR transaction to block chain with (receiving address, email, md5, project, total credits done), to be confirmed by other 
>> * UserB (Verifier) receives POR block
>> * User verifies e-mail -> recieving address from gridcoin.us
>> * User gets XML from project using email/md5 and verifies work done
>> * User confirms block

You stated in your previous post that UserA signs the POR block with the email/md5, then the verifier verifies the email using the emal/md5. I'm getting mixed messages here.

**Comment by gregagnew** [comment]
Okay I've deleted all my previous and obsolete comments. Is that clear?

**Comment by Foggyx420** [comment]
the email is used for the CPID and many instances in code for using of it. it is not public. sure team founder can access that but with removal of team req a user doesn't have to be in a team for it to be shown.

**Comment by Erkan-Yilmaz** [comment]
* a related poll: [Gridcoin poll: Impact to GRC price if team requirement is lifted? (vote until July 25)](https://steemit.com/gridcoin/@erkan/gridcoin-poll-impact-to-grc-price-if-team-requirement-is-lifted-vote-until-july-25)
* see also: "irc chat about biased answers on this poll" (#434)

**Comment by fffffgggg54** [comment]
What statistics does the network have access to? If I were to look up an account on a stats website, is that all the network can see?

**Comment by fffffgggg54** [comment]
So the current "main" problem is establishing proof of CPID ownership, correct?
If so, could we not just interface with the BOINC Client as the BOINC Client would know, right?
Sorry if I'm not getting some of the tech and underlying infrastructure right.
